packages/starlight/components/Header.astro

```diff
@@ -8,14 +9,20 @@ import {
 	SocialIcons,
 	ThemeSelect,
 } from 'virtual:starlight/components';
+
+/**
+ * Render the `Search` component if Pagefind is enabled or the default search component has been overridden.
```

Comment:
Should this behavior referenced somewhere like in the search component override reference?

Reply:
Ah, maybe. I initially thought about disabling rendering inside `<Search>` itself, but that would result in some CSS & JS still being loaded even if the UI isn’t, so moved it here. I guess I assumed if you override `Search` you don’t also want your override to be hidden if `pagefind` is disabled.

Reply:
Added a note in accf0a2!